### PR TITLE
Media Browser: Use Media Class

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change
67945970b31ab8d399f69f63314e6b44

201856da38b59667e2cc9feca7f22fae

## Type of change

- [x] New feature (thank you!)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

### CODE REVIEW

File: src/components/media-player/ha-media-player-browse.ts [hunk 1]
```diff
@@ -263,13 +228,19 @@ export class HaMediaPlayerBrowse extends LitElement {
...
+        ? mediaClass.layout === "grid"
           ? html`
-              <div class="children">
+              <div
+                class="children ${classMap({
+                  portrait: mediaClass.thumbnail_ratio === "portrait",
+                })}"
+              >
                 ${currentItem.children.map(
                   (child) => html`
```
Comment (bramkragten) [inline src/components/media-player/ha-media-player-browse.ts:245]:
```suggestion
                  (child) => { const classStyles = MediaClassBrowserSettings[child.media_class]; return html`
```

Comment (zsarnett) [reply src/components/media-player/ha-media-player-browse.ts:245]: Not sure what you are going for here

Comment (zsarnett) [reply src/components/media-player/ha-media-player-browse.ts:247]: There are no per child classes needed

**Comment by zsarnett** [comment]
Need to fix safari rendering in this PR as well

**Comment by zsarnett** [comment]
Also I  should fix this https://github.com/home-assistant/frontend/issues/6946

**Comment by zsarnett** [comment]
Going to fix this too https://github.com/home-assistant/frontend/issues/6949 EZ fix

**Comment by zsarnett** [comment]
@bramkragten I cant figure out the damn safari bug right now. Also, its late and Friday. and beers so

File: src/components/media-player/ha-media-player-browse.ts [hunk 2]
```diff
@@ -504,14 +491,38 @@ export class HaMediaPlayerBrowse extends LitElement {
...
+            href="https://www.home-assistant.io/integrations/media_source/#local-media"
+            target="_blank"
+            rel="noreferrer"
+            >documentation</a
+          >
+          for more info
+        </p>
+      `;
+    }
+    return err.message;
```
Comment (bramkragten) [inline src/components/media-player/ha-media-player-browse.ts:523]:
```suggestion
    return html`<span class="error">err.message</span>`;
```

File: src/components/media-player/ha-media-player-browse.ts [hunk 3]
```diff
@@ -286,19 +264,24 @@ export class HaMediaPlayerBrowse extends LitElement {
...
                         >
-                          ${child.can_expand && !child.thumbnail
+                          ${!child.thumbnail
                             ? html`
                                 <ha-svg-icon
                                   class="folder"
-                                  .path=${mdiFolder}
+                                  .path=${MediaClassBrowserSettings[
+                                    child.children_media_class ||
+                                      child.media_class
```
Comment (bramkragten) [inline src/components/media-player/ha-media-player-browse.ts:273]: We should only look at the `children_media_class` for directory/library
```suggestion
                                    child.children_media_class ||
                                      child.media_class
```

File: src/components/media-player/ha-media-player-browse.ts [hunk 4]
```diff
@@ -678,11 +696,16 @@ export class HaMediaPlayerBrowse extends LitElement {
         }
 
         .child .play {
+          transition: color 0.5s;
+          border-radius: 50%;
+          bottom: calc(50% - 24px);
+          right: calc(50% - 24px);
```
Comment (bramkragten) [inline src/components/media-player/ha-media-player-browse.ts:702]: Oh, this was an experiment, not happy with it yet. Would like the old behavior when it is not expandable, show the play button on hover.

File: src/components/media-player/ha-media-player-browse.ts (hunk 1 above)
Comment (bramkragten) [reply src/components/media-player/ha-media-player-browse.ts:252]: You keep looking up the `MediaClassBrowserSettings`, do it once and store in a var.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**